packages/api/src/controllers/stream.js

```diff
@@ -1233,7 +1233,10 @@ async function terminateStreamReq(req, stream) {
 
 const streamDetectionEvent = "stream.detection";
 
-app.post("/hook", async (req, res) => {
+app.post("/hook", authMiddleware({}), async (req, res) => {
+  if (!req.user.admin && !req.user.access?.broadcasterHooks) {
```

Comment:
Can we move this into `auth-middleware`, similar to `anyAdmin` flag?

Reply:
The problem with that would be how to represent the "admin OR broadcasterHooks" idea in the params object. For example something like `{ anyAdmin: true, access: { broadcasterHooks: true } }` could make sense for this specific case, but it could also look like that _both_ things need to be set instead, and not one or another. 

I could make a more generic interface in the auth middleware, like receiving a function that validates properties on the `User` object (and token type?) and returns a `boolean` instead. Then already provide some helper implementations like `isAdmin`, `isAnyAdmin` and `isAnyAdminOrBroadcaster` which can be set on the params objects. WDYT?

Reply:
Implemented the idea we agreed offline :)